src/main/java/core/basesyntax/utils/transaction/handlers/OperationHandler.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.utils.transaction.handlers;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+
+public interface OperationHandler {
+    void perform(FruitTransaction transaction, StorageImpl storage);
```

Comment:
According to the SOLID principles, and more specifically the Dependency Inversion Principle, we should depend on abstractions and not on concrete classes. Here, the `OperationHandler` interface method `perform` should accept an abstraction of `Storage` rather than the concrete `StorageImpl`. Consider using the `Storage` interface as the method argument type instead.